sound/soc/sof/intel/hda.c

```diff
@@ -472,10 +476,37 @@ static int hda_init_caps(struct snd_sof_dev *sdev)
 	else
 		dev_info(bus->dev, "hda codecs found, mask %lx!\n", bus->codec_mask);
 
-	/* used by hda machine driver to create dai links */
-	mach_params = (struct snd_soc_acpi_mach_params *)
-		&sdev->pdata->machine->mach_params;
-	mach_params->codec_mask = bus->codec_mask;
+	if (bus->codec_mask) {
+		for (i = 0; i < HDA_MAX_CODECS; i++) {
+			if (bus->codec_mask & (1 << i))
+				codec_num++;
+		}
+
+		/* if there are HDMI codec and HDA codecs, hda machine
+		 * is applied. For I2S codec + HDMI codec, there is only one
+		 * hda codec. For other cases, the value should be zero.
+		 */
+		if (codec_num > 1) {
```

Comment:
The logic is wrong. Is there is a single codec, the skl_hda_generic_machine driver can work
this should be codec_num > 0

Reply:
yes, one case